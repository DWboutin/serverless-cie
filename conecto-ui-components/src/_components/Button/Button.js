import React from 'react'
import PropTypes from 'prop-types'
import omit from 'omit.js'
import classNames from 'classnames'

class Button extends React.Component {
  constructor(props) {
    super(props)

    this.classesPrefix = 'to-btn'
  }

  render() {
    const {
      className,
      disabled,
      htmlType,
      handleClick,
      comp,
      active,
    } = this.props
    const classes = classNames({
      [this.classesPrefix]: true,
      active: active,
      'is-disabled': disabled,
      [`${this.classesPrefix}-${this.props.type}`]: this.props.type,
      [`${className}`]: className,
    })
    const componentProps = omit(this.props, [
      'comp',
      'handleClick',
      'htmlType',
      'active',
    ])

    return React.createElement(comp, {
      ...componentProps,
      type: comp === 'button' ? htmlType : '',
      onClick: handleClick,
      className: classes,
    })
  }
}

Button.defaultProps = {
  comp: 'button',
  htmlType: 'button',
  type: 'primary',
}

Button.propTypes = {
  comp: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.func,
    PropTypes.symbol,
  ]),
  active: PropTypes.bool,
  htmlType: PropTypes.string,
  type: PropTypes.oneOf([
    'primary',
    'bordered',
    'ghost',
    'underlined',
    'menu',
    'switch',
    'filter',
    'account',
    'roofing',
  ]),
  className: PropTypes.string,
  disabled: PropTypes.bool,
  handleClick: PropTypes.func,
}

export default Button
