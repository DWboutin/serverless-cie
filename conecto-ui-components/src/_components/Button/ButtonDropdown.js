import React from 'react'
import PropTypes from 'prop-types'
import omit from 'omit.js'
import classNames from 'classnames'
import closest from '../_helpers/closest'
import closeDropdown from '../_helpers/closeDropdown'

class Button extends React.Component {
  constructor(props) {
    super(props)

    this.classesPrefix = 'to-btn'

    this.state = {
      active: false,
    }

    this.handleClick = this.handleClick.bind(this)
    this.handleClickOutside = this.handleClickOutside.bind(this)
  }

  componentDidMount() {
    const { id } = this.props

    if (typeof window.TO_DROPDOWN_LISTENER === 'undefined') {
      window.TO_DROPDOWN_LISTENER = true
      window.TO_DROPDOWN_IDS = {
        [`to-dropdown-${id}`]: this.handleClickOutside,
      }

      document.addEventListener('click', closeDropdown, false)
    } else {
      window.TO_DROPDOWN_IDS[`to-dropdown-${id}`] = this.handleClickOutside
    }
  }

  componentWillUnmount() {
    const { id } = this.props

    delete window.TO_DROPDOWN_IDS[`to-dropdown-${id}`]

    if (Object.keys(window.TO_DROPDOWN_IDS).length === 0) {
      delete window.TO_DROPDOWN_LISTENER
      document.removeEventListener('click', closeDropdown, false)
    }
  }

  handleClickOutside(e) {
    if (
      this.state.active === true &&
      closest(e.target, '.to-btn-dropdown-wrapper') === null
    ) {
      this.setState({
        active: false,
      })
    }
  }

  handleClick(e) {
    const { handleClick } = this.props

    if (
      closest(e.target, 'div').className.indexOf('to-btn-dropdown-wrapper') !==
      -1
    ) {
      this.setState(
        {
          active: !this.state.active,
        },
        () => {
          if (handleClick) {
            handleClick(e)
          }
        }
      )
    } else {
      e.preventDefault()
    }
  }

  render() {
    const { className, htmlType, comp, links } = this.props
    const { active } = this.state
    const wrapperClasses = classNames({
      'to-btn-dropdown-wrapper': true,
      active: active,
    })
    const classes = classNames({
      [this.classesPrefix]: true,
      'to-btn-dropdown': true,
      active: active,
      [`${this.classesPrefix}-${this.props.type}`]: this.props.type,
      [`${className}`]: className,
    })
    const componentProps = omit(this.props, [
      'comp',
      'handleClick',
      'htmlType',
      'active',
      'type',
      'links',
    ])

    return (
      <div className={wrapperClasses}>
        {React.createElement(comp, {
          ...componentProps,
          type: comp === 'button' ? htmlType : '',
          onClick: this.handleClick,
          className: classes,
        })}
        {links && links.length > 0 && (
          <div className="to-btn-dropdown-menu">
            <ul>
              {links.map((link, index) => (
                <li key={`btn-drop-${index}`}>{link}</li>
              ))}
            </ul>
          </div>
        )}
      </div>
    )
  }
}

Button.defaultProps = {
  comp: 'button',
  htmlType: 'button',
  type: 'primary',
}

Button.propTypes = {
  id: PropTypes.string.isRequired,
  comp: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.func,
    PropTypes.symbol,
  ]),
  active: PropTypes.bool,
  htmlType: PropTypes.string,
  type: PropTypes.oneOf([
    'primary',
    'bordered',
    'ghost',
    'underlined',
    'menu',
    'account',
    'switch',
    'filter',
    'roofing',
  ]),
  className: PropTypes.string,
  handleClick: PropTypes.func,
  links: PropTypes.array.isRequired,
}

export default Button
