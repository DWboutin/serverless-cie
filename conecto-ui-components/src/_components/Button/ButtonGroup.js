import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

class ButtonGroup extends React.Component {
  constructor(props) {
    super(props)

    this.classesPrefix = 'to-btn-group'
  }

  render() {
    const { className } = this.props
    const classes = classNames({
      [this.classesPrefix]: true,
      [`${className}`]: className,
    })

    return <div className={classes}>{this.props.children}</div>
  }
}

ButtonGroup.defaultProps = {}

ButtonGroup.propTypes = {
  children: PropTypes.arrayOf(PropTypes.element).isRequired,
}

export default ButtonGroup
