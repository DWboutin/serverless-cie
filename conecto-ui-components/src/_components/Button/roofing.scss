@import '../_base/colors';

.to-btn-group {
  display: inline-block;
  position: relative;

  .to-btn {
    &:first-child {
      border-top-right-radius: 0px;
      border-bottom-right-radius: 0px;

      &.to-btn-roofing {
        border-right: 1px solid lighten($roofing, 10%);
      }
    }

    &:not(:first-child):not(:last-child) {
      border-radius: 0;

      &.to-btn-roofing {
        border-right: 1px solid lighten($roofing, 10%);
        border-left: 1px solid darken($roofing, 10%);
      }
    }

    &:last-child {
      border-top-left-radius: 0px;
      border-bottom-left-radius: 0px;

      &.to-btn-roofing {
        border-left: 1px solid darken($roofing, 10%)
      }
    }
  }
}

.to-btn {
  &-roofing {
    color: #ffffff;
    background-color: $roofing;
    box-shadow: 0px 0px 0px rgba($roofing, 0);

    &::before {
      box-shadow: 0px 6px 8px rgba($roofing, 0.25);
    }
  }
}