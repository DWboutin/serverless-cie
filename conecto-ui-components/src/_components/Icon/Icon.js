import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import To from '../Logo/Logos/To'
import Menu from './Icons/Menu'
import Alert from './Icons/Alert'
import ArrowDown from './Icons/ArrowDown'
import Bookmark from './Icons/Bookmark'
import Check from './Icons/Check'
import Close from './Icons/Close'
import Cog from './Icons/Cog'
import Concurrents from './Icons/Concurrents'
import Email from './Icons/Email'
import Facebook from './Icons/Facebook'
import Flag from './Icons/Flag'
import ColoredFlag from './Icons/ColoredFlag'
import Marker from './Icons/Marker'
import Notification from './Icons/Notification'
import Phone from './Icons/Phone'
import Search from './Icons/Search'
import SMS from './Icons/SMS'
import ThumbUp from './Icons/ThumbUp'
import User from './Icons/User'

import CCAmex from './Icons/CCAmex'
import CCDefault from './Icons/CCDefault'
import CCDinersClub from './Icons/CCDinersClub'
import CCDiscover from './Icons/CCDiscover'
import CCJCB from './Icons/CCJCB'
import CCMastercard from './Icons/CCMastercard'
import CCVisa from './Icons/CCVisa'

class Icon extends React.Component {
  constructor(props) {
    super(props)

    this.classesPrefix = 'to-icon'
  }

  iconDefinition(type) {
    switch (type) {
      case 'alert':
        return Alert
      case 'arrow-down':
        return ArrowDown
      case 'bookmark':
        return Bookmark
      case 'check':
        return Check
      case 'close':
        return Close
      case 'cog':
        return Cog
      case 'concurrents':
        return Concurrents
      case 'email':
        return Email
      case 'facebook':
        return Facebook
      case 'flag':
        return Flag
      case 'colored-flag':
        return ColoredFlag
      case 'marker':
        return Marker
      case 'menu':
        return Menu
      case 'notification':
        return Notification
      case 'phone':
        return Phone
      case 'search':
        return Search
      case 'sms':
        return SMS
      case 'thumb-up':
        return ThumbUp
      case 'user':
        return User
      case 'cc-amex':
        return CCAmex
      case 'cc-default':
        return CCDefault
      case 'cc-dinersclub':
        return CCDinersClub
      case 'cc-discover':
        return CCDiscover
      case 'cc-jcb':
        return CCJCB
      case 'cc-mastercard':
        return CCMastercard
      case 'cc-visa':
        return CCVisa
      case 'conecto':
      default:
        return To
    }
  }

  render() {
    const { className, type } = this.props
    const classes = classNames({
      [this.classesPrefix]: true,
      [`${className}`]: className,
    })
    const iconToRender = this.iconDefinition(type)

    return (
      <i className={classes} aria-label={`icon: ${type}`}>
        {React.createElement(iconToRender)}
      </i>
    )
  }
}

Icon.defaultProps = {
  type: 'conecto',
}

const IconTypePropTypes = PropTypes.oneOf([
  'alert',
  'arrow-down',
  'bookmark',
  'check',
  'close',
  'cog',
  'concurrents',
  'email',
  'facebook',
  'flag',
  'colored-flag',
  'marker',
  'menu',
  'notification',
  'phone',
  'search',
  'sms',
  'thumb-up',
  'user',
  'cc-amex',
  'cc-default',
  'cc-dinersclub',
  'cc-discover',
  'cc-jcb',
  'cc-mastercard',
  'cc-visa',
  'conecto',
])

Icon.propTypes = {
  type: IconTypePropTypes,
  className: PropTypes.string,
}

export { Icon as default, IconTypePropTypes }
