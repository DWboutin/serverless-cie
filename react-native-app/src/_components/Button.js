import React from 'react'
import { StyleSheet, Text } from 'react-native'

import TouchableItem from './TouchableItem'

const Button = ({ children, onPress, type }) => {
  const buttonStyle = StyleSheet.flatten([styles.button, styles[type]])

  return (
    <TouchableItem style={buttonStyle} onPress={onPress}>
      <Text style={styles[`${type}Text`]} onPress={onPress}>
        {children}
      </Text>
    </TouchableItem>
  )
}

const styles = StyleSheet.create({
  button: {
    paddingTop: 12,
    paddingRight: 20,
    paddingBottom: 2,
    paddingLeft: 20,
    height: 35,
  },
  primary: {
    backgroundColor: '#26B89F',
    borderRadius: 30,
  },
  bordered: {
    backgroundColor: 'transparent',
    borderWidth: 1,
    borderColor: '#572D2D',
    borderRadius: 15,
    paddingTop: 10,
    height: 30,
  },
  ghost: {
    backgroundColor: '#fff',
    borderWidth: 0,
    borderRadius: 15,
    paddingTop: 10,
    height: 30,
  },
  underlined: {
    paddingLeft: 0,
    paddingRight: 0,
    borderBottomWidth: 2,
    borderBottomColor: '#572D2D',
  },
  disabled: {
    backgroundColor: '#26B89F',
    borderRadius: 30,
    opacity: 0.25,
  },
  primaryText: {
    color: '#ffffff',
    fontSize: 14,
    fontFamily: 'sailec-regular',
  },
  borderedText: {
    color: '#572D2D',
    fontSize: 12,
    fontFamily: 'sailec-regular',
  },
  ghostText: {
    color: '#572D2D',
    fontSize: 12,
    fontFamily: 'sailec-regular',
  },
  underlinedText: {
    color: '#572D2D',
    fontSize: 10,
    fontFamily: 'sailec-regular',
  },
  disabledText: {
    color: '#ffffff',
    fontSize: 14,
    fontFamily: 'sailec-regular',
  },
})

Button.defaultProps = {
  type: 'primary',
  onPress: () => {},
}

export default Button
