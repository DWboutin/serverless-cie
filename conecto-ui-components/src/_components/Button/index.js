import Button from './Button'
import ButtonGroup from './ButtonGroup'
import ButtonDropdown from './ButtonDropdown'

export { Button as default, ButtonGroup, ButtonDropdown }
