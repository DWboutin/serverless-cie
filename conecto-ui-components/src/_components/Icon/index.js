import Icon, { IconTypePropTypes } from './Icon'

export { Icon as default, IconTypePropTypes }
