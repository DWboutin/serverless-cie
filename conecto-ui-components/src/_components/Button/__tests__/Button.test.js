import React from 'react'
import { shallow, mount } from 'enzyme'
import { Link } from 'gatsby'

import Button from '../../Button'
import Icon from '../../Icon'

global.console = {
  error: jest.fn(),
}

describe('Button', () => {
  describe('Primary', () => {
    it('should match snapshot', () => {
      const wrapper = shallow(<Button>Test</Button>)

      expect(wrapper).toMatchSnapshot()
    })

    it('should have the prefix class', () => {
      const wrapper = shallow(<Button>Test</Button>)

      expect(wrapper.instance().classesPrefix).toEqual('to-btn')
      expect(wrapper.hasClass(wrapper.instance().classesPrefix)).toEqual(true)
    })

    it('should contain added classes', () => {
      const wrapper = shallow(
        <Button className="test-class-1 test-class-2">Test</Button>
      )

      expect(wrapper.hasClass('test-class-1')).toEqual(true)
      expect(wrapper.hasClass('test-class-2')).toEqual(true)
    })

    it('should render an Html button with good props', () => {
      const wrapper = shallow(<Button>Test</Button>)

      expect(wrapper.text()).toEqual('Test')
      expect(wrapper.find('button').length).toEqual(1)
      expect(wrapper.find('button').props().type).toEqual('button')
      expect(wrapper.find('button').props().className).toEqual(
        'to-btn to-btn-primary'
      )
    })

    it('should match snapshot with an Icon', () => {
      const wrapper = shallow(
        <Button comp={Link}>
          <Icon />
          Test
        </Button>
      )

      expect(wrapper).toMatchSnapshot()
    })

    it('should render an Icon correctly', () => {
      const wrapper = mount(
        <Button comp={Link}>
          <Icon />
          Test
        </Button>
      )

      expect(wrapper.find(Icon).length).toEqual(1)

      wrapper.unmount()
    })
  })

  describe('Bordered', () => {
    it('should match snapshot', () => {
      const wrapper = shallow(<Button type="bordered">Test</Button>)

      expect(wrapper).toMatchSnapshot()
    })

    it('should have the prefix class', () => {
      const wrapper = shallow(<Button type="bordered">Test</Button>)

      expect(wrapper.instance().classesPrefix).toEqual('to-btn')
      expect(wrapper.hasClass(wrapper.instance().classesPrefix)).toEqual(true)
    })

    it('should render an Html button with good props', () => {
      const wrapper = shallow(<Button type="bordered">Test</Button>)

      expect(wrapper.text()).toEqual('Test')
      expect(wrapper.find('button').length).toEqual(1)
      expect(wrapper.find('button').props().type).toEqual('button')
      expect(wrapper.find('button').props().className).toEqual(
        'to-btn to-btn-bordered'
      )
    })
  })

  describe('Ghost', () => {
    it('should match snapshot', () => {
      const wrapper = shallow(<Button type="ghost">Test</Button>)

      expect(wrapper).toMatchSnapshot()
    })

    it('should have the prefix class', () => {
      const wrapper = shallow(<Button type="ghost">Test</Button>)

      expect(wrapper.instance().classesPrefix).toEqual('to-btn')
      expect(wrapper.hasClass(wrapper.instance().classesPrefix)).toEqual(true)
    })

    it('should render an Html button with good props', () => {
      const wrapper = shallow(<Button type="ghost">Test</Button>)

      expect(wrapper.text()).toEqual('Test')
      expect(wrapper.find('button').length).toEqual(1)
      expect(wrapper.find('button').props().type).toEqual('button')
      expect(wrapper.find('button').props().className).toEqual(
        'to-btn to-btn-ghost'
      )
    })
  })

  describe('Underlined', () => {
    it('should match snapshot', () => {
      const wrapper = shallow(<Button type="underlined">Test</Button>)

      expect(wrapper).toMatchSnapshot()
    })

    it('should have the prefix class', () => {
      const wrapper = shallow(<Button type="underlined">Test</Button>)

      expect(wrapper.instance().classesPrefix).toEqual('to-btn')
      expect(wrapper.hasClass(wrapper.instance().classesPrefix)).toEqual(true)
    })

    it('should render an Html button with good props', () => {
      const wrapper = shallow(<Button type="underlined">Test</Button>)

      expect(wrapper.text()).toEqual('Test')
      expect(wrapper.find('button').length).toEqual(1)
      expect(wrapper.find('button').props().type).toEqual('button')
      expect(wrapper.find('button').props().className).toEqual(
        'to-btn to-btn-underlined'
      )
    })
  })

  describe('Switch', () => {
    it('should match snapshot', () => {
      const wrapper = shallow(<Button type="switch">Test</Button>)

      expect(wrapper).toMatchSnapshot()
    })

    it('should have the prefix class', () => {
      const wrapper = shallow(<Button type="switch">Test</Button>)

      expect(wrapper.instance().classesPrefix).toEqual('to-btn')
      expect(wrapper.hasClass(wrapper.instance().classesPrefix)).toEqual(true)
    })

    it('should render an Html button with good props', () => {
      const wrapper = shallow(<Button type="switch">Test</Button>)

      expect(wrapper.text()).toEqual('Test')
      expect(wrapper.find('button').length).toEqual(1)
      expect(wrapper.find('button').props().type).toEqual('button')
      expect(wrapper.find('button').props().className).toEqual(
        'to-btn to-btn-switch'
      )
    })
  })

  describe('Filter', () => {
    it('should match snapshot', () => {
      const wrapper = shallow(<Button type="filter">Test</Button>)

      expect(wrapper).toMatchSnapshot()
    })

    it('should have the prefix class', () => {
      const wrapper = shallow(<Button type="filter">Test</Button>)

      expect(wrapper.instance().classesPrefix).toEqual('to-btn')
      expect(wrapper.hasClass(wrapper.instance().classesPrefix)).toEqual(true)
    })

    it('should render an Html button with good props', () => {
      const wrapper = shallow(<Button type="filter">Test</Button>)

      expect(wrapper.text()).toEqual('Test')
      expect(wrapper.find('button').length).toEqual(1)
      expect(wrapper.find('button').props().type).toEqual('button')
      expect(wrapper.find('button').props().className).toEqual(
        'to-btn to-btn-filter'
      )
    })
  })

  describe('Menu', () => {
    it('should match snapshot', () => {
      const wrapper = shallow(<Button type="menu">Test</Button>)

      expect(wrapper).toMatchSnapshot()
    })

    it('should match snapshot with the menu icon', () => {
      const wrapper = shallow(
        <Button type="menu">
          test
          <Icon type="menu" />
        </Button>
      )

      expect(wrapper).toMatchSnapshot()
    })

    it('should have the prefix class', () => {
      const wrapper = shallow(<Button type="menu">Test</Button>)

      expect(wrapper.instance().classesPrefix).toEqual('to-btn')
      expect(wrapper.hasClass(wrapper.instance().classesPrefix)).toEqual(true)
    })

    it('should render an Html button with good props', () => {
      const wrapper = shallow(<Button type="menu">Test</Button>)

      expect(wrapper.text()).toEqual('Test')
      expect(wrapper.find('button').length).toEqual(1)
      expect(wrapper.find('button').props().type).toEqual('button')
      expect(wrapper.find('button').props().className).toEqual(
        'to-btn to-btn-menu'
      )
    })
  })

  describe("With Gatsby's Link component", () => {
    it('should match snapshot', () => {
      const wrapper = shallow(<Button comp={Link}>Test</Button>)

      expect(wrapper).toMatchSnapshot()
    })

    it('should render a Link component correctly', () => {
      const wrapper = shallow(<Button comp={Link}>Test</Button>)

      expect(wrapper.find(Link).length).toEqual(1)
    })

    it('should render a Link tag with good props', () => {
      const wrapper = mount(<Button comp={Link}>Test</Button>)

      expect(wrapper.find('button').length).toEqual(0)
      expect(wrapper.find('a').length).toEqual(1)
      expect(wrapper.find('a').hasClass('to-btn to-btn-primary')).toEqual(true)

      wrapper.unmount()
    })
  })

  it('should console.error when not passing a string or a React component to comp prop', () => {
    shallow(<Button comp={123}>Test</Button>)

    expect(global.console.error).toBeCalled()
  })

  it('should console.error when not passing a string to type prop', () => {
    shallow(<Button type={123}>Test</Button>)

    expect(global.console.error).toBeCalled()
  })

  it('should console.error when not passing a string to htmlType prop', () => {
    shallow(<Button htmlType={123}>Test</Button>)

    expect(global.console.error).toBeCalled()
  })

  it('should console.error when not passing a function to handleClick prop', () => {
    shallow(<Button handleClick={123}>Test</Button>)

    expect(global.console.error).toBeCalled()
  })
})
