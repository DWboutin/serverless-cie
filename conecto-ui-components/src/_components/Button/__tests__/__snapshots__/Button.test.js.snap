// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Bordered should match snapshot 1`] = `
<button
  className="to-btn to-btn-bordered"
  type="button"
>
  Test
</button>
`;

exports[`Button Filter should match snapshot 1`] = `
<button
  className="to-btn to-btn-filter"
  type="button"
>
  Test
</button>
`;

exports[`Button Ghost should match snapshot 1`] = `
<button
  className="to-btn to-btn-ghost"
  type="button"
>
  Test
</button>
`;

exports[`Button Menu should match snapshot 1`] = `
<button
  className="to-btn to-btn-menu"
  type="button"
>
  Test
</button>
`;

exports[`Button Menu should match snapshot with the menu icon 1`] = `
<button
  className="to-btn to-btn-menu"
  type="button"
>
  test
  <Icon
    type="menu"
  />
</button>
`;

exports[`Button Primary should match snapshot 1`] = `
<button
  className="to-btn to-btn-primary"
  type="button"
>
  Test
</button>
`;

exports[`Button Primary should match snapshot with an Icon 1`] = `
<mockConstructor
  className="to-btn to-btn-primary"
  type=""
>
  <Icon
    type="conecto"
  />
  Test
</mockConstructor>
`;

exports[`Button Switch should match snapshot 1`] = `
<button
  className="to-btn to-btn-switch"
  type="button"
>
  Test
</button>
`;

exports[`Button Underlined should match snapshot 1`] = `
<button
  className="to-btn to-btn-underlined"
  type="button"
>
  Test
</button>
`;

exports[`Button With Gatsby's Link component should match snapshot 1`] = `
<mockConstructor
  className="to-btn to-btn-primary"
  type=""
>
  Test
</mockConstructor>
`;
